app/hooks/StrapiAPI.ts

```diff
@@ -56,6 +55,32 @@ export function usePageContent(title: string) {
   );
 }
 
+export function useNavigationData() {
+  const path = "header?populate[logo]=*&populate[navigation][populate]=*";
+
+  const dataFetcher = () =>
+    fetcher<{ data: StrapiApi.StrapiDatumResponse<StrapiApi.HeaderResponse> }>(
+      `${config.STRAPI_API_URL}/api/${path}`,
+      {
+        Authorization: `Bearer ${config.STRAPI_API_TOKEN}`,
+      }
+    );
+
+  const { data, error, isLoading } = <
+    {
+      data: { data: StrapiApi.StrapiDatumResponse<StrapiApi.HeaderResponse> };
+      error: any;
+      isLoading: boolean;
+    }
+  >useSWR(`/api/${path}`, dataFetcher);
+
+  return {
+    data: data?.data ? data?.data.attributes : null,
+    error,
+    isLoading,
+  };
+}
```

Comment:
@seanmalbert you're right that with multiple return types `useStrapiHook` requires extra downstream extraction and type work that pulls the concern out of this adapter layer. We could likely get fancier with even more layers of generics and type narrowing but it's definitely not worth the effort now. My suggestion to go this route was definitely premature given we're rapidly onboarding Strapi. 

So rewind, here I'm encapsulating all the particulars for the `/header` endpoint in one function.

Reply:
I'm glad you went this route now, it definitely feels like this will be easier to maintain and scale.